5-data-modeling/event-manifest-populator/run.py

```diff
@@ -0,0 +1,281 @@
+
+
+def split_full_path(path):
+    """Return pair of bucket without protocol and path
+    Arguments:
+    path - valid S3 path, such as s3://somebucket/events
+    >>> split_full_path('s3://mybucket/path-to-events')
+    ('mybucket', 'path-to-events/')
+    >>> split_full_path('s3://mybucket')
+    ('mybucket', None)
+    """
+    if path.startswith('s3://'):
+        path = path.lstrip('s3://')
+    elif path.startswith('s3n://'):
+        path = path.lstrip('s3n://')
+    else:
+        raise ValueError("S3 path should start with s3:// or s3n:// prefix")
+    parts = path.split('/')
+    bucket = parts[0]
+    path = '/'.join(parts[1:])
+    return (bucket, normalize_prefix(path))
+
+
+def normalize_prefix(path):
+    """Add trailing slash to prefix if it is not present
+    >>> normalize_prefix("somepath")
+    'somepath/'
+    >>> normalize_prefix("somepath/")
+    'somepath/'
+    """
+    if path is None or path is '' or path is '/':
+        return None
+    elif path.endswith('/'):
+        return path
+    else:
+        return path + '/'
+
+
+def run(command):
+    output = subprocess.Popen(command.split(),
+                              stdout=subprocess.PIPE,
+                              stderr=subprocess.PIPE)
+    (stdout, stderr) = output.communicate()
+    print(stdout)
+    if output.returncode != 0:
+        sys.exit(stderr)
+
+
+def validate_input_path(s3_path):
+    if not s3_path.startswith("s3://"):
+        raise ValueError("Invalid enriched archive path. It should start with s3:// protocol")
+
+
+def validate_since(since):
+    try:
+        datetime.strptime(since, SINCE_SHORT_FORMAT)
+    except ValueError:
+        try:
+            datetime.strptime(since, SINCE_PRECISE_FORMAT)
+        except ValueError:
+            print("Incorrect --since format. Should conform YY-MM-dd or YY-MM-dd-HH-mm-ss format")
+            sys.exit(1)
+
+
+def get_valid_region(location):
+    return S3_LOCATIONS[location]
+
+
+def base64encode(file):
+    content = open(file, 'r').read()
+    if sys.version_info >= (3, 0):
+        content = str.encode(content)
+    return base64.urlsafe_b64encode(content)
+
+
+def test(args):
+    print("Running 'sbt test'. This may take few minutes")
+    run('sbt test')
+
+
+def assembly(args):
+    print("Running 'sbt assembly'. This may take few minutes")
+    run('sbt assembly')
+
+
+def upload(args):
+    (bucket, path) = split_full_path(args.jar_bucket)
+    c = boto.connect_s3(profile_name=args.profile)
+    b = c.get_bucket(bucket)
+
+    k2 = Key(b)
+    if path is None:
+        k2.key = JAR_FILE
+    else:
+        k2.key = path + JAR_FILE
+    k2.set_contents_from_filename(DIR_WITH_JAR + JAR_FILE)
+
+
+def run_emr(args):
+    validate_input_path(args.enriched_archive)
+    if args.since is not None:
+        validate_since(args.since)
+        since_arg = ["--since", args.since]
+    else:
+        since_arg = []
+
+    c = boto.connect_s3(profile_name=args.profile)
```

Comment:
Interesting. Nothing was working for me on Ubuntu. I'll rerun later this morning when there's a new version of this script and see what happens...